### PR TITLE
[SYCL][Joint Matrix] Add test for multiple elem-wise ops

### PR DESCRIPTION
**Pull request by YuriPlyakhin**
Add new test for multiple elem-wise ops for different matrix types in the same kernel

### CODE REVIEW

**Comment by YuriPlyakhin** [comment]
Failures:
AMD
```
Failed Tests (3):
  SYCL :: Basic/built-ins/host_math.cpp
  SYCL :: Basic/get_backend.cpp
  SYCL :: Regression/device_num.cpp
```
CUDA (tagging @intel/llvm-reviewers-cuda):
```
Failed Tests (7):
  SYCL :: Basic/built-ins/host_math.cpp
  SYCL :: Basic/get_backend.cpp
  SYCL :: Regression/device_num.cpp
  SYCL :: Regression/invalid_reqd_wg_size_correct_exception.cpp
  SYCL :: Tracing/usm/queue_copy_released_pointer.cpp
  SYCL :: Tracing/usm/queue_single_task_nullptr.cpp
  SYCL :: Tracing/usm/queue_single_task_released_pointer.cpp
```
are unrelated, as this PR just adds one more e2e test. Also noticed these same tests failing in other PR (#10259)

**Comment by YuriPlyakhin** [comment]
@yubingex007-a11y , please, review

**Review by yubingex007-a11y — APPROVED** [review]

**Comment by YuriPlyakhin** [comment]
@intel/llvm-gatekeepers, please merge

**Comment by aelovikov-intel** [comment]
> @intel/llvm-gatekeepers, please merge

We can't merge PRs with two weeks old test results.

File: sycl/test-e2e/Matrix/element_wise_abc_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  size_t NDRangeN = N / TN;
+  buffer<T2, 2> bufA(A.get_data(), range<2>(M, K));
+  buffer<T2, 2> bufB(B.get_data(), range<2>(K, N));
+  buffer<T1, 2> bufC(C.get_data(), range<2>(M, N));
+
+  queue q;
+  q.submit([&](handler &cgh) {
+     accessor accC{bufC, cgh, read_write};
+     accessor accA{bufA, cgh, read_write};
+     accessor accB{bufB, cgh, read_write};
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Matrix/element_wise_abc_impl.hpp:43]:
```suggestion
     accessor accC{bufC, cgh};
     accessor accA{bufA, cgh};
     accessor accB{bufB, cgh};
```

File: sycl/test-e2e/Matrix/element_wise_abc_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+         [=](nd_item<2> spmd_item) [[intel::reqd_sub_group_size(SG_SZ)]] {
+           // The submatrix API has to be accessed by all the workitems in a
+           // subgroup these functions will be called once by the subgroup no
+           // code divergence between the workitems
+           const auto global_idx = spmd_item.get_global_id(0);
+           const auto global_idy = spmd_item.get_global_id(1);
+           const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+           ext::oneapi::sub_group sg = spmd_item.get_sub_group();
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Matrix/element_wise_abc_impl.hpp:56]: Can we use `sycl::sub_group` instead? I think they might be just aliases in our implementation.

File: sycl/test-e2e/Matrix/element_wise_abc_impl.hpp (hunk 1 above)
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/element_wise_abc_impl.hpp]: Fixed

File: sycl/test-e2e/Matrix/element_wise_abc_impl.hpp (hunk 2 above)
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/element_wise_abc_impl.hpp]: Fixed

**Comment by YuriPlyakhin** [comment]
@intel/llvm-gatekeepers , please, merge. All green.

**Pull request merged**